### ISSUE TITLE
1 2 3 = one zero zero?

### ISSUE THREAD

**Issue by Tarcaxoxide** [Tarcaxoxide/Sauce#12]
I don't get what is going on [here](https://github.com/Tarcaxoxide/Sauce/blob/9d4a007fb0e3cbd14696feb6f2f34e437a5b258b/kernel/src/Graphics/Shell/Langauge/Exec.cpp#L16).

the first "word" is interpreted correctly but proceeding "words" are all zero??

**Comment by paulstelian97** [comment]
The lexer looks a tiny bit sketch, did you check that the tokens themselves are fine?

**Comment by Tarcaxoxide** [comment]
> The lexer looks a tiny bit sketch, did you check that the tokens themselves are fine?

ya, they should be fine i think?, i did print them out again at some point and they print correctly.
that's what i currently have at the moment, just have it printing out the tokens with newlines between them.

if i had to guess at the problem (just barely a guess though) i think something might be getting "optimized" in a less than ideal way.
me and the compiler have had that kind of problem in the past XD.

**Comment by Tarcaxoxide** [comment]
it's either that or i've managed to screw up memory management again somehow without realizing it.
memory is still kind of feels like witchcraft to me right now.

**Comment by Tarcaxoxide** [comment]
tried to print the tokens to serial and it just freezes the entire system.
indeed there seems to be a deeper problem here but not entirely sure what exactly that is at the moment.

**Comment by paulstelian97** [comment]
Do you have an assembly output for the function itself? If you blame the optimizer maybe the result lies in there.

**Comment by Tarcaxoxide** [comment]
no, i don't know how to do that. but i've had similar if not exactly the same kind of thing running on my linux host just fine.
it's usually not the Lexer i fail on but the Parser and Evaluator.

though at this point i might try just to rewrite the entire system at some point since i've already been planning on switching out gnu-efi for anything with an MIT license to match the rest of my code anyways.

**Comment by Tarcaxoxide** [comment]
eh figured out how :) deer lord it looks quite a bit more messy in assembly XD i almost feel sorry for the assembler.

**Comment by paulstelian97** [comment]
Yeah the reason I ask for assembly output is that if things are weird there you've likely encountered undefined behaviour and the optimizer has exploited some assumption that you violated. We could talk all day about the specific instance to figure out how to do things right (I'm a fan of doing it right first, fast only as needed)

**Comment by Tarcaxoxide** [comment]
I'm not really for fast necessarily however i can only do so much for correctness given my lack of proper education in the field.

all i can do is stumble around in the dark and hope i don't fall over that cliff that i can't see.
honestly surprised i was able to get as far as i have before running to something like this.

I've compiled all the compile units into assembly files along with the compilation of the object files and have packed them into a tar file.
I'll go a head and do a TR and attach them. I already look at what i believed to be the correct assemble file but i don't understand enough to even know what it's doing.

**Comment by paulstelian97** [comment]
Let's just say that the assembly isn't easy to use.

See if you can make the compiler generate assembly with C/C++ lines interspersed as comments. It might be something that can be done with -O0 builds.

.o files compiled with -g would also be good because those can allow me to obtain that and more info. If I can see how the lines of assembler map to the lines of C++ I can see if something maps in an unexpected way.

**Comment by Tarcaxoxide** [comment]
TR5 should contain assembly files with comments now and the object files should be compiled with -g as this file [lib.tar.gz](https://github.com/Tarcaxoxide/Sauce/releases/download/TR5/lib.tar.gz)

and i already compile with '-O0' but the required flag was '-fverbose-asm' apparently and so i've added that.
the and the '-g' flag
